server/src-lib/Hasura/GraphQL/Execute/Query.hs

```diff
@@ -40,55 +36,56 @@ import           Hasura.EncJSON
 import           Hasura.GraphQL.Context
 import           Hasura.GraphQL.Execute.Action
 import           Hasura.GraphQL.Execute.Prepare
+import           Hasura.GraphQL.Execute.Remote
 import           Hasura.GraphQL.Execute.Resolve
 import           Hasura.GraphQL.Parser
 import           Hasura.Prelude
 import           Hasura.RQL.DML.RemoteJoin
 import           Hasura.RQL.DML.Select                  (asSingleRowJsonResp)
 import           Hasura.RQL.Types
 import           Hasura.Session
--- import           Hasura.SQL.Types
 import           Hasura.SQL.Value
 
 import qualified Hasura.RQL.DML.Select                  as DS
 
-data PGPlan
-  = PGPlan
-  { _ppQuery       :: !Q.Query
-  , _ppVariables   :: !PlanVariables
-  , _ppPrepared    :: !PrepArgMap
-  , _ppRemoteJoins :: !(Maybe RemoteJoins)
+data PreparedSql
+  = PreparedSql
+  { _psQuery       :: !Q.Query
+  , _psPrepArgs    :: !PrepArgMap
+  , _psRemoteJoins :: !(Maybe RemoteJoins)
   }
+  deriving Show
 
-instance J.ToJSON PGPlan where
-  toJSON (PGPlan q vars prepared _) =
-    J.object [ "query"     J..= Q.getQueryText q
-             , "variables" J..= vars
-             , "prepared"  J..= fmap show prepared
+-- | Required to log in `query-log`
+instance J.ToJSON PreparedSql where
+  toJSON (PreparedSql q prepArgs _) =
+    J.object [ "query" J..= Q.getQueryText q
+             , "prepared_arguments" J..= fmap (pgScalarValueToJson . snd) prepArgs
              ]
 
+-- | The computed SQL with alias which can be logged. Nothing here represents no
+-- SQL for cases like introspection responses. Tuple of alias to a (maybe)
+-- prepared statement
```

Comment:
This comment says this is a “tuple,” but this is a map. I’ve revised the comment to what I think is correct, but feel free to change my wording if I’m wrong or if you have something better.

```suggestion
-- | Maps root fields to the SQL used to execute them. For fields that do not
-- generate any SQL, the value is 'Nothing'. This information is retained in
-- this form only for logging; it is not consulted during actual execution.
```